src/NFTBill.sol

```diff
@@ -39,6 +40,35 @@ contract NFTBill is ERC1155 {
         _mint(msg.sender, id, 1, '');
     }
 
+    function depositWithPermit(
+        address erc20,
+        uint96 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
+        if (value <= 0) revert ValueTooSmall();
+
+        IERC20Permit(erc20).permit(
+            msg.sender,
+            address(this),
+            value,
+            deadline,
+            v,
+            r,
+            s
+        );
+        ERC20(erc20).transferFrom(msg.sender, address(this), value);
+        // TODO: What if we get less tokens than we asked for?
+        uint256 id;
+        assembly {
+            id := or(value, shl(96, erc20))
+        }
+
+        _mint(msg.sender, id, 1, '');
+    }
```

Comment:
what if this just calls `deposit(erc20, value)`? Could simplify things a bit, even if it costs marginally more

Reply:
thanks for the suggestion! updated in the latest [commit](https://github.com/enclosed-money/contracts/pull/13/commits/9f395c422541f567bb57f64b4d0bbee07396066d) :+1: